ui/app/AppLayouts/Chat/ContactsColumn/Channel.qml

```diff
@@ -178,6 +170,28 @@ Rectangle {
             color: Style.current.white
         }
     }
+
+    MouseArea {
+        cursorShape: Qt.PointingHandCursor
+        acceptedButtons: Qt.LeftButton | Qt.RightButton
+        anchors.fill: parent
+        hoverEnabled: true
+        onEntered: {
+          wrapper.hovered = true
+        }
+        onExited: {
+          wrapper.hovered = false
+        }
+        onClicked: {
+            if (mouse.button & Qt.RightButton) {
+                channelContextMenu.openMenu(index, muted, chatType, name, chatId, identicon)
+                return;
+            }
+            chatsModel.setActiveChannelByIndex(index)
+            chatGroupsListView.currentIndex = index
+        }
+    }
+
```

Comment:
The reason this needed to be moved at the end of this component is so that we ensure the MouseArea covers all children of the wrapping Rectangle (otherwise it wouldn't apply to e.g. the message preview area. Weird, I know.